### ISSUE TITLE
Inheritable JsonKey with Freezed

### ISSUE THREAD

**Issue by torbenkeller** [google/json_serializable.dart#885]
This issue is related to https://github.com/rrousselGit/freezed/issues/423

**Is your feature request related to a problem? Please describe.**
I'm building an API wrapper, and I'm dealing with two JSON object that have numerous shared keys and only a few specific ones. I have an interface like this to implement this:

```dart
abstract class MyBaseType {
  @JsonKey(name: 'property_key')
  final String property;
}
```

And a couple of freezed classes like this:

```dart
@freezed
class MySpecializedType extends MyBaseType with _$MySpecializedType {
  const MySpecializedType._();

  const factory MySpecializedType({
    required String property,
    @JsonKey(name: 'another_propery_key') required String anotherProperty,
  }) = _MySpecializedType
}
```

Normally, when using json_serializable, the JsonKey annotation in the base class will be used in subclasses. This is not happening when using freezed.

**Describe the solution you'd like**
I'd like JsonKey annotations used in superclasses to be used in freezed subclasses as well.

**Describe alternatives you've considered**
The only alternative is writing the same annotation everywhere it's needed.

**Comment by kevmoo** [comment]
This gets complicated, sadly. It's hard to pick a consistent ordering of fields to search first.

**Comment by mpiparo** [comment]
related...
With latest v5.0.0 (or perhaps I'm just noticing), I'm getting an annotation warning on 'JsonKey' when annotating in a freezed class:

"The annotation 'JsonKey' can only be used on fields or gettersdart(invalid_annotation_target)"

```dart
@freezed
class UserProfileDto with _$UserProfileDto {
  factory UserProfileDto({
    @JsonKey(ignore: true) @Default('') String id,
    required String name,
    required String userBio,
    required List<PhotoDto> profilePhotosList,
    required int numPosts,
    required int numMatches,
    @TimestampConverterAuto() required Option<Timestamp> modifyDate,
  }) = _UserProfileDto;
```

**Comment by kevmoo** [comment]
Does `freezed` use `JsonKey`?

`json_serializabel` DOES NOT support annotating constructor params.

**Comment by mpiparo** [comment]
No, looks like it just passes the annotation thru to its generated code (below), which is working ok.  But what you're saying makes sense, since json_serializable doesn't support annotating on constructors, the warning is valid then. I just hadn't seen that warning before.  

Also looks like there's a thread on [freezed issues](https://github.com/rrousselGit/freezed/issues/476) on this topic. 

```dart
class _$UserProfileDtoTearOff {
  const _$UserProfileDtoTearOff();

  _UserProfileDto call(
      {@JsonKey(ignore: true) String id = '',
      required String name,
      required String userBio,
      required List<PhotoDto> profilePhotosList,
      required int numPosts,
...
```

**Comment by kevmoo** [comment]
The warning IS new as of the latest release of pkg:json_annotation

**Comment by Ngochiendev** [comment]
same issue, Have you found a solution for the warning The annotation 'JsonKey' can only be used on fields or getters
